File: src/main/java/pl/tsimur/jpatest/service/UserService.java
package pl.tsimur.jpatest.service;

import pl.tsimur.jpatest.model.Tariff;
import pl.tsimur.jpatest.model.User;
import pl.tsimur.jpatest.model.dto.UserMiniDto;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

public interface UserService {

    User save(User user);

    List<User> saveAll(List<User> users);

    Optional<User> find(Integer id);

    List<User> findAll();

    List<User> findAll(String name);

    List<User> findAll(List<Integer> ids);

    List<UserMiniDto> findAllMiniUsers();

    List<User> findAllOlder(Integer age);

    List<User> findAllYounger(Integer age);

    List<User> findAllToUpdatePurchase();

    List<User> findAllUsersWithEnoughMoney(LocalDateTime localDateTime);

    List<User> findAllWithSubs();

    Integer findCountEmployeesWithPensionerAge(String orgName);

    User findOrThrow(Integer id);

    void updateName(Integer id, String name);

    void buySubscription(Integer id, Tariff tariff);

    void updateMonthlyPayment();

    void updateBalance(Integer id, Float balance);

    void updatePurchaseDate(Integer id, LocalDateTime date);

    void delete(Integer id);

    void delete();

}

File: src/main/kotlin/pl/timur/jpatest/service/UserService.kt
package pl.timur.jpatest.service

import pl.timur.jpatest.model.Tariff
import pl.timur.jpatest.model.User
import pl.timur.jpatest.model.dto.UserMiniDto
import java.time.LocalDateTime

interface UserService {
    fun save(user: User): User
    fun saveAll(users: List<User>): List<User>
    fun find(id: Int): User?
    fun findAll(): List<User>
    fun findAll(name: String): List<User>
    fun findAll(ids: List<Int>): List<User>
    fun findAllMiniUsers(): List<UserMiniDto>
    fun findAllOlder(age: Int): List<User>
    fun findAllYounger(age: Int): List<User>
    fun findAllToUpdatePurchase(): List<User>
    fun findAllUsersWithEnoughMoney(localDateTime: LocalDateTime): List<User>
    fun findAllWithSubs(): List<User>
    fun findCountEmployeesWithPensionerAge(orgName: String): Int
    fun findOrThrow(id: Int): User
    fun updateName(id: Int, name: String)
    fun buySubscription(id: Int, tariff: Tariff)
    fun updateMonthlyPayment()
    fun updateBalance(id: Int, balance: Float)
    fun updatePurchaseDate(id: Int, date: LocalDateTime)
    fun delete(id: Int)
    fun delete()
}